### PR TITLE
Cachegroups localization policy

### PR DESCRIPTION
**Pull request by rawlinp**
Provide the ability to specify which localization methods are allowed by a cachegroup when routing a client.

The typical process goes:
1. client makes a request to TR (DNS or HTTP)
2. TR "localizes" that client's IP address using the following methods: Deep CZF (if enabled for the DS), CZF, then GEO (via GeoIP Database like MaxMind).
3. TR selects a cachegroup to route the client to using the result from step 2

This provides the ability during step 3 to filter out potential cachegroups that aren't enabled for the localization method used. For instance, if you had a cachegroup that was not directly connected to the internet, you might want to use that cachegroup for only Deep CZF and CZF clients. Off-net clients (which would be localized via GEO), would *not* be routed to that particular cachegroup. This configuration would be achieved by setting `"localizationMethods": ["CZ", "DEEP_CZ"]` for that cachegroup.

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2151/
Test PASSed.

File: traffic_ops/testing/api/v13/tc-fixtures.json [hunk 1]
```diff
@@ -38,6 +38,11 @@
             "parentCachegroupName": "parentCachegroup",
             "secondaryParentCachegroupName": "secondaryCachegroup",
             "shortName": "cg1",
+            "localizationMethods": [
```
Comment (dewrich) [inline traffic_ops/testing/api/v13/tc-fixtures.json]: Shouldn't these just be the .String() func call on your `LocalizationMethod`?

**Comment by dewrich** [comment]
@rawlinp API Test Case fails:

```DEBUG: cachegroups_test.go:33: 2018-07-31T09:47:58.735331-06:00: TestCacheGroupsAuthentication() PASSED:
--- FAIL: TestCacheGroups (0.76s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1399a48]

goroutine 67 [running]:
testing.tRunner.func1(0xc4201801e0)
	/usr/local/Cellar/go@1.9/1.9.7/libexec/src/testing/testing.go:711 +0x2d2
panic(0x1434c00, 0x1748d10)
	/usr/local/Cellar/go@1.9/1.9.7/libexec/src/runtime/panic.go:491 +0x283
github.com/apache/trafficcontrol/traffic_ops/testing/api/v13.UpdateTestCacheGroups(0xc4201801e0)
	/Users/dricha209/projects/go/src/github.com/apache/trafficcontrol/traffic_ops/testing/api/v13/cachegroups_test.go:177 +0xd28
github.com/apache/trafficcontrol/traffic_ops/testing/api/v13.TestCacheGroups(0xc4201801e0)
	/Users/dricha209/projects/go/src/github.com/apache/trafficcontrol/traffic_ops/testing/api/v13/cachegroups_test.go:34 +0x63
```

File: traffic_ops/testing/api/v13/tc-fixtures.json (hunk 1 above)
Comment (rawlinp) [reply traffic_ops/testing/api/v13/tc-fixtures.json]: Do you mean pre-processing/injecting this into the JSON using the String() methods before loading it for the tests?

**Comment by dewrich** [comment]
@rawlinp yea Jeremy did because there's a bug in it, we should prioritize it

File: traffic_ops/testing/api/v13/tc-fixtures.json (hunk 1 above)
Comment (dewrich) [reply traffic_ops/testing/api/v13/tc-fixtures.json]: Just referring to the enum value instead of hard coding

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2186/
Test PASSed.

**Comment by rawlinp** [comment]
just re-ran the Go API integration tests post-rebase, and they are passing now that the Go cachegroups API has been reenabled: 
```
PASS
ok  	github.com/apache/trafficcontrol/traffic_ops/testing/api/v13	53.988s
```

**Pull request merged**